test/classes/initializers/errors/errHandling/deinitFields.future

```diff
@@ -0,0 +1,8 @@
+unimplemented feature: throwing initializers don't deinit fields
+
+This test notes that when an initializer throws, we don't do anything
+to deinit the object's fields, which it seems we must do to clean up.
+Here, I've set up the .good assuming we would not call C's deinit()
+because C was never fully initialized.
```

Comment:
This last comment is the only major sticking point for me. If we are in `postinit`, isn't the object fully initialized such that it is fair game to `deinit`?

Reply:
... which is not the case for `init`, certainly.

Reply:
(I am also realizing that this future is about `init`)

Reply:
> If we are in postinit, isn't the object fully initialized such that it is fair game to deinit?

Applying that question to the other future (`test/classes/initializers/postInit/deinitFields.chpl`)… that's a really good question and probably one that we haven't answered (at least that I can recall, and not surprisingly since postinits haven't been able to throw).  I think it depends on whether we consider the object fully initialized once we get through all the inits, or whether it includes the postinits as well.

I could imagine the name `postinit` suggesting "initialization is done", though it could also simply mean "after the inits are called" similar to how code after `init this;` is after the [current static type's] fields have been initialized, yet still potentially involved in initializing the object overall.

Let me update the [other] .future and its issue to note this question without taking a stance on it.

Reply:
Thanks! This is probably more about us clarifying rules and expectations around these functions more than anything else. I think codifying something along the lines of "calling `deinit` is legal immediately after `init` finishes" sounds reasonable, and in my mind implies that it can be called before or during `postinit`.

Thinking more on this, I don't know what's legal in `postinit`. Can `postinit` deinitialize a field and then reinitialize? Can it throw in between those two? Could `deinit` handle that case?

Reply:
We don't permit calls to `deinit()` explicitly, so the type author couldn't deinit a field from within a postinit in that sense.  Of course, they probably know what it would take to manually deinit and reinit the field; but then, arguably they could do what is necessary to get it into an appropriate state before throwing anyway?  (that is, prepare it for the deinit call or take the deinit-ing actions to it themselves if need be).

Reply:
Yeah, that's a good point.